### PR TITLE
Implement batch APIs of IAdapter interface

### LINKED ISSUE #30: Batch API is still missing for Casbin.NET
See: https://casbin.org/docs/en/overview#feature-set-for-different-languages

This adapter needs to implement the IAdapter interface with `AddPolicies()` and `RemovePolicies()` functions.

This issue is to track the progress of solving: https://github.com/casbin-net/EFCore-Adapter/issues/28

### PR DESCRIPTION
**Pull request by Sagilio**
fix #30 and #31

### CODE REVIEW

**Review by hsluoyz — APPROVED** [review]

**Comment by hsluoyz** [comment]
@xcaptain plz review.

File: EFCore-Adapter/CasbinDbAdapter.cs [hunk 1]
```diff
@@ -133,24 +144,81 @@ public virtual void AddPolicy(string section, string policyType, IList<string> r
...
+                return;
+            }
+
+            var rulesArray = rules as IList<string>[] ?? rules.ToArray();
+            if (rulesArray.Length is 0)
+            {
+                return;
+            }
+
+            var casbinRules = rulesArray.Select(r =>
```
Comment (xcaptain) [inline EFCore-Adapter/CasbinDbAdapter.cs:171]: Should we avoid adding duplicate rules into the database? If a rule already exists, should we add it twice? Duplicate rules may not affect the enforce result, so you may ignore this comment if you think it's ok.

**Review by xcaptain — APPROVED** [review]
Left a small comment, the overall pull request looks good to me.

File: EFCore-Adapter/CasbinDbAdapter.cs (hunk 1 above)
Comment (Sagilio) [reply EFCore-Adapter/CasbinDbAdapter.cs:171]: I think we should avoid it and I will recommit later.

Comment (Sagilio) [reply EFCore-Adapter/CasbinDbAdapter.cs:171]: I think the duplicate rules should be removed on InternalEnforcer at Casbin.NET, we need not avoid this problem at Adapter.

Comment (Sagilio) [reply EFCore-Adapter/CasbinDbAdapter.cs:171]: In my opinion, the duplicate rules should be removed on InternalEnforcer at Casbin.NET, we need not avoid this problem at Adapter.

**Pull request merged**